tools/dds/dds-sniffer/rs-dds-sniffer.h

```diff
@@ -10,55 +10,87 @@
 
 #include <librealsense2/dds/dds-participant.h>
 
+#include <fastdds/dds/subscriber/Subscriber.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
+#include <fastdds/dds/core/status/SubscriptionMatchedStatus.hpp>
+
 class dds_sniffer
 {
 public:
     dds_sniffer();
     ~dds_sniffer();
 
-    bool init( librealsense::dds::dds_domain_id domain = 0, bool snapshot = false, bool machine_readable = false );
+    bool init( librealsense::dds::dds_domain_id domain = 0, bool snapshot = false, bool machine_readable = false,
+               bool topic_samples = false );
     void run( uint32_t seconds );
 
 private:
     std::shared_ptr< librealsense::dds::dds_participant::listener > _listener;
     librealsense::dds::dds_participant _participant;
 
-    std::map< librealsense::dds::dds_guid, std::string> _discovered_participants;
+    // For listing entities in the domain
+    std::map< librealsense::dds::dds_guid, std::string > _discovered_participants;
     struct topic_info
     {
         std::set< librealsense::dds::dds_guid > readers;
         std::set< librealsense::dds::dds_guid > writers;
     };
-    std::map< std::string, topic_info > _discovered_topics;
+    std::map< std::string, topic_info > _topics_info_by_name;
```

Comment:
If it maps to `topic_info` then shouldn't it be `_topic_info_by_name`?

Reply:
Since the map stores more than one `topic_info` I used plural tense.
Using `_topic_infos_by_name` seemed off, because it is the `topic` name not the `topic_info` name